pkg/systemd/init.js

```diff
@@ -823,4 +829,484 @@ define([
+        } else if (timer_unit.repeat["index"] === 1440) {
+            timer_unit.OnCalendar = repeat_array.map(function (item) {
+                return "OnCalendar=*-*-* " + Number(item.hours) + ":" + Number(item.minutes) + ":00";
+            });
+        } else if (timer_unit.repeat["index"] === 10080) {
+            timer_unit.OnCalendar = repeat_array.map(function (item) {
+                return "OnCalendar=" + item.days_text.slice(0,3) + " *-*-* " + Number(item.hours) + ":" + Number(item.minutes) + ":00";
+            });
+        } else if (timer_unit.repeat["index"] === 44640) {
+            timer_unit.OnCalendar = repeat_array.map(function (item) {
+                return "OnCalendar=*-*-" + item.days_value + " " + Number(item.hours) + ":" + Number(item.minutes) + ":00";
+            });
+        } else if (timer_unit.repeat["index"] === 525600) {
+            timer_unit.OnCalendar = repeat_array.map(function (item) {
+                return "OnCalendar=*-" + moment(item.date_to_parse).format('MM') + "-" + moment(item.date_to_parse).format('DD') + " " +Number(item.hours) + ":" + Number(item.minutes) + ":00";
+            });
+        }
+        if (timer_unit.repeat["index"] !== 60)
+            timer_unit.OnCalendar = timer_unit.OnCalendar.toString().replace(/,/g,"\n");
+        var invalid = create_timer_file();
+        if (invalid)
+            return false;
+        init_units();
+        return true;
+    }
+    function create_timer_file() {
+        var unit = "[Unit]\nDescription=";
+        var service = "\n[Service]\nExecStart=";
+        var timer = "\n[Timer]\n";
+        var service_file = unit + timer_unit.Description + service + timer_unit.Command + "\n[Install]\nWantedBy=default.target";
+        var timer_file = " ";
+        if (timer_unit.Calendar_or_Boot == "Boot") {
+            var boottimer = timer +"OnBootSec=" + timer_unit.boot_time + timer_unit.boot_time_unit;
+            timer_file = unit + timer_unit.Description + boottimer;
+        }
+        else if (timer_unit.Calendar_or_Boot == "Calendar") {
+            var calendartimer = timer + timer_unit.OnCalendar;
+            timer_file = unit + timer_unit.Description + calendartimer;
+        }
+        // writing to file
+        var service_path = "/etc/systemd/system/" + timer_unit.name + ".service";
+        var file = cockpit.file(service_path, { superuser: 'try' });
+        file.replace(service_file).
+            fail(function(error) {
+                console.log(error);
+            });
+        console.log("#Service file#\n" + service_file);
+        var timer_path = "/etc/systemd/system/" + timer_unit.name + ".timer";
+        file = cockpit.file(timer_path, { superuser: 'try' });
+        file.replace(timer_file).
+            fail(function(error) {
+                console.log(error);
+            });
+        console.log("#Timer file#\n" + timer_file);
```

Comment:
@dperpeet before we merge this, we should remove these _console.log()_ lines. They give the contents of .service and .timer files.